components/esp_lvgl_port/CMakeLists.txt

```diff
@@ -22,12 +22,21 @@ endif()
 
 set(PORT_PATH "src/${PORT_FOLDER}")
 
+# Include SIMD assembly source code for rendering
+if(CONFIG_IDF_TARGET_ESP32 OR CONFIG_IDF_TARGET_ESP32S3)
+    if (lvgl_ver VERSION_GREATER_EQUAL "9.0.0")
+        message(VERBOSE "Compiling SIMD")
+        file(GLOB_RECURSE ASM_SOURCES ${PORT_PATH}/simd/*.S)
+    endif()
+endif()
+
 idf_component_register(
-        SRCS "${PORT_PATH}/esp_lvgl_port.c" "${PORT_PATH}/esp_lvgl_port_disp.c" 
+        SRCS "${PORT_PATH}/esp_lvgl_port.c" "${PORT_PATH}/esp_lvgl_port_disp.c" ${ASM_SOURCES}
         INCLUDE_DIRS "include" 
         PRIV_INCLUDE_DIRS "priv_include"
         REQUIRES "esp_lcd" 
-        PRIV_REQUIRES "esp_timer")
+        PRIV_REQUIRES "esp_timer"
+        WHOLE_ARCHIVE)
```

Comment:
Without the `WHOLE_ARCHIVE` the project would not link. In order to remove the `WHOLE_ARCHVIE` we can for example use the same approach as in [esp-usb](https://github.com/espressif/esp-usb/blob/c50d3c8511d39bc0e00f92e6477b2cd35cf92215/host/class/uvc/usb_host_uvc/test_app/main/CMakeLists.txt#L24) to force link an .S file, which would require to force link each .S file, since there is only one function present in each .S file. And those functions do no call each other.

@igrr is there any other way of not not using `WHOLE_ARCHIVE` in  this case?